retail/api/grs-search/SearchWithQueryExpansion.cs

```diff
@@ -0,0 +1,119 @@
+        }
+
+        // Get search service request
+        private static SearchRequest GetSearchRequest(string query, SearchRequest.Types.QueryExpansionSpec.Types.Condition condition)
+        {
+            var queryExpansionSpec = new SearchRequest.Types.QueryExpansionSpec()
+            {
+                Condition = condition
+            };
+            var searchRequest = new SearchRequest()
+            {
+                Placement = DefaultSearchPlacement, // Placement is used to identify the Serving Config name
+                Query = query,
+                QueryExpansionSpec = queryExpansionSpec,
+                VisitorId = "123456", // A unique identifier to track visitors
+                PageSize = 10
+            };
+
+            var jsonSerializeSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore,
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                Formatting = Formatting.Indented
+            };
+
+            var searchRequestJson = JsonConvert.SerializeObject(searchRequest, jsonSerializeSettings);
+
+            Console.WriteLine("\nSearch. request: \n\n" + searchRequestJson);
+            return searchRequest;
+        }
+
+        // Call the Retail Search:
+        [Attributes.Example]
+        public static IEnumerable<SearchResponse> Search() // PagedEnumerable<SearchResponse, SearchResponse.Types.SearchResult>
+        {
+            // TRY DIFFERENT QUERY EXPANSION CONDITION HERE:
+            var condition = SearchRequest.Types.QueryExpansionSpec.Types.Condition.Auto;
+            string query = "Google Youth Hero Tee Grey";
+
+            var searchRequest = GetSearchRequest(query, condition);
+
+            var searchResponse = GetSearchServiceClient().Search(searchRequest).AsRawResponses(); //.AsRawResponses();
+
+            Console.WriteLine("\nSearch. response: \n");
+
+            var jsonSerializeSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore,
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                Formatting = Formatting.Indented
+            };
+
+            var firstSearchResponse = searchResponse.FirstOrDefault();
+
+            if (firstSearchResponse != null)
+            {
+                var objectToSerialize = new
+                {
+                    results = firstSearchResponse.Results,
+                    totalSize = firstSearchResponse.TotalSize,
+                    attributionToken = firstSearchResponse.AttributionToken,
+                    nextPageToken = firstSearchResponse.NextPageToken,
+                    facets = firstSearchResponse.Facets,
+                    queryExpansionInfo = firstSearchResponse.QueryExpansionInfo
+                };
+
+                var serializedJson = JsonConvert.SerializeObject(objectToSerialize, jsonSerializeSettings);
```

Comment:
Again please don't do this.

If the sample wants to print some of the resulting items and some of the properties of the operation result please do so with code similar to:

```Console.WriteLine(response.NextPageToken)```

Reply:
We want to show our users the search response in the output. If we just print the raw response, it is not readable at all, so we decided to use pretty printed JSON